src/tests/base_test.py

```diff
@@ -78,6 +82,41 @@ def update_parameterized(cls, request: pytest.FixtureRequest, config: BaseConfig
                 else:
                     raise AttributeError(f"No attribute name {fixture_name} in {config_type} object type")
 
+    @pytest.fixture
+    def k8s_minikube(self, request: pytest.FixtureRequest) -> kubernetes.dynamic.DynamicClient:
+        """k8 minikube wrapper client  object fixture passed to test function
+        This function fixture called by testcase, we initialize k8client
+        Storing configmap before test begins -> for later restoring
+        When test func done, check if configmap was changed and restore config.
+        :return:
+        """
+        log.debug(f"--- SETUP --- Creating k8s_minikube for test: {request.node.name}\n")
+        k8_client_ = kubernetes.dynamic.DynamicClient(
+            kubernetes.client.api_client.ApiClient(
+                configuration=kubernetes.config.load_kube_config(),
+            ),
+        )
+        minikube = MinikubeController(k8_client_)
+        # before test begins verify all developments are ready
+        if not minikube.verify_deployments_are_ready():
+            assert "Minikube deployments services not ready/active"
+        configmap_before = dict(minikube.configmap_data['data'])
+        yield minikube
+        log.debug(f"--- TEARDOWN --- Deleting k8s_minikube resources for test: {request.node.name}\n")
+        configmap_after = dict(minikube.configmap_data['data'])
+        # Detect if configmap data was changed - need to restore configuration
+        if configmap_before != configmap_after:
+            log.debug(f"configmap {minikube.configmap_name} was changed during the test"
+                      f" {request.node.name}, restoring\n")
+            minikube.patch_resource("ConfigMap",
+                                    "assisted-service-config", {'data': configmap_before})
+            log.info(f"Set deployment scale assisted-service to 0 for test {request.node.name} -waiting")
+            minikube.change_deployment_scale("assisted-service", 0)
+            log.info(f"Set deployment scale assisted-service to 1 for test {request.node.name} -waiting")
+            minikube.change_deployment_scale("assisted-service", 1)
```

Comment:
BTW, the clean way to rollout the deployment is to set an annotation: https://github.com/kubernetes-client/python/issues/1378

Reply:
Ack the reason i used scale options because in resiliency testing i must to stop the service and have full controll when in scale up it again ... in addition it allow me to place with scale of assisted-pods if one day we will have more then one.  i implemented the waiters. if no difference in function we can use it unless scale down does not work ...  we can add for sure this option too.

Reply:
> maybe you can factorize this code in `MinikubeController` into a method `rollout_assisted_service`

done